app/src/main/java/io/apicurio/registry/content/canon/XsdContentCanonicalizer.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.content.canon;
+
+import java.io.IOException;
+
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.apache.xml.security.Init;
+import org.apache.xml.security.c14n.CanonicalizationException;
+import org.apache.xml.security.c14n.Canonicalizer;
+import org.apache.xml.security.c14n.InvalidCanonicalizerException;
+import org.xml.sax.SAXException;
+
+import io.apicurio.registry.content.ContentCanonicalizer;
+import io.apicurio.registry.content.ContentHandle;
+
+/**
+ * @author cfoskin@redhat.com
+ */
+public class XsdContentCanonicalizer implements ContentCanonicalizer {
+
+    /**
+     * @see io.apicurio.registry.content.ContentCanonicalizer#canonicalize(io.apicurio.registry.content.ContentHandle)
+     */
+    @Override
+    public ContentHandle canonicalize(ContentHandle content) {
+        Init.init(); // this library needs to be initialized using the static init()
+        try {
+            Canonicalizer canon = Canonicalizer.getInstance(Canonicalizer.ALGO_ID_C14N_OMIT_COMMENTS);
+            byte[] canonXmlBytes = canon.canonicalize(content.bytes());
+            String canonicalized = new String(canonXmlBytes, "UTF-8"); // for UTF-8 encoding
```

Comment:
Use `java.nio.charset.StandardCharsets.UTF_8` here instead of the string.

Reply:
👍

Reply:
No, use IoUtil ... so we have bytes <--> string all in one single place!

Reply:
@EricWittmann @alesj I can use ```String canonicalized = IOUtils.toString(canon.canonicalize(content.bytes()), "UTF-8");
``` but I cant use the `StandardCharsets.UTF_8` then? it either takes a `byte[], String` or `InputStream, Charset`   wdyt?

Reply:
Just use `toString()` on the charset then:  `StandardCharsets.UTF_8.toString()`.

Reply:
I think Ales is suggesting you use `IoUtil` which is a registry class.  No the one from apache commons.

Reply:
https://github.com/Apicurio/apicurio-registry/blob/master/common/src/main/java/io/apicurio/registry/utils/IoUtil.java#L109-L111

Reply:
ah.. will update